java/vortex-jni/src/main/java/dev/vortex/arrow/ArrowAllocation.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.arrow;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+
+public final class ArrowAllocation {
+    private static final RootAllocator ROOT_ALLOCATOR = new RootAllocator(Long.MAX_VALUE);
+
+    private ArrowAllocation() {}
+
+    public static BufferAllocator rootAllocator() {
```

Comment:
Somewhat confusingly, this references the `org.apache.arrow` types, but this is the subproject that gets shaded, so consumers of this will see this as the shaded `dev.vortex.relocated.org.apache.arrow.memory.BufferAllocator`.

Everything in vortex-spark project works with the shaded types